crates/bevy_gizmos/src/light.rs

```diff
@@ -0,0 +1,181 @@
+//! A module adding debug visualization of f [`PointLight`]s, [`SpotLight`]s and [`DirectionalLight`]s.
+
+use std::f32::consts::PI;
+
+use crate::{self as bevy_gizmos, primitives::dim3::GizmoPrimitive3d};
+
+use bevy_app::{Plugin, PostUpdate};
+use bevy_color::LinearRgba;
+use bevy_ecs::{
+    component::Component,
+    query::{With, Without},
+    reflect::ReflectComponent,
+    schedule::IntoSystemConfigs,
+    system::{Query, Res},
+};
+use bevy_math::{
+    primitives::{Cone, Sphere},
+    Quat, Vec3,
+};
+use bevy_pbr::{DirectionalLight, PointLight, SpotLight};
+use bevy_reflect::{std_traits::ReflectDefault, Reflect};
+use bevy_transform::{components::GlobalTransform, TransformSystem};
+
+use crate::{
+    config::{GizmoConfigGroup, GizmoConfigStore},
+    gizmos::Gizmos,
+    AppGizmoBuilder,
+};
+
+fn draw_gizmos<'a, P, S, D>(
+    point_lights: P,
+    spot_lights: S,
+    directional_lights: D,
+    gizmos: &mut Gizmos<LightGizmoConfigGroup>,
+) where
```

Comment:
Nit: It might be nice from an organizational/readability perspective to split these up into a function-per-light